test/caching.bats

```diff
@@ -182,3 +182,25 @@ EOF
     umoci unpack --image oci:mode-test dest
     [ -x dest/rootfs/executable ]
 }
+
+@test "can read previous version's cache" {
+    git clone https://github.com/anuvu/stacker
+    (cd stacker && make)
+
+    # note that this is not quite a complete test: since the cache has a
+    # types.Layer in it, anything that can change in there could break this. so
+    # really, any time we update that structure, we should update this test.
```

Comment:
Could you expand on this point about updating this test? What would need updating?
It's also not obvious (to me) how this tests the cache layer format?  Without digging into
the harness, how do we know what cache dir the stacker command is using and what
version of the cache is in-use?

Reply:
> Could you expand on this point about updating this test? What would need updating?

Sure, mostly it's that if you add a new attribute to layer, you should add it here so that we test that we can read+write it across cache versions. I should probably add something for everything that exists in layer right now, although I was trying to cook up a way to reflectively add these things in a test so it wouldn't be so manual.

> It's also not obvious (to me) how this tests the cache layer format? Without digging into
the harness, how do we know what cache dir the stacker command is using and what
version of the cache is in-use?

The cache version is written into the cache file itself by stacker. Since the test builds the github.com/anuvu/stacker master branch, it will always be the "old" cache version if the patch in question changes the cache version. The subsequent build is with the "new" version on whatever branch the test is being run, so if the cache can't be parsed by the new code, things will fail (modulo the point above about being incomplete).

Each test is run in its own subdirectory, so the cache dir is just the default .stacker for both invocations.

Reply:
Ok, I added a reflection-based test that doesn't need all the caveats about "you have to remember to update this" to try and catch this problem in the future. I left this test here because it doesn't really hurt anything, but I dropped the comment about needing to update it.